### PR TITLE
Consider group valid if `(is_required)` set to `false`

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  1) Changes the structure of the project renaming Java-related modules to reside under the `java-tests` subproject.
  2) Adds tests for vanilla-java code generation (under the new module called `vanilla`).
  3) Fixes the edge case when the option `(is_required)` is present but set to `false`.

It also carries out the test for `(goes_with)`, but its full satisfaction is left to the one of the following PRs.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, оно-то сейчас билдится. Но используется версия валидации, которая встроена в ProtoData. А как заставить работать ту, что сейчас в разработке в таком варианте, я пока придумать не могу.

**Comment by alexander-yevsyukov** [comment]
@armiol, посмотрел в сгенерированный код модуля `consumer`, там работает — буду разбираться.

**Comment by alexander-yevsyukov** [comment]
> @armiol, посмотрел в сгенерированный код модуля `consumer`, там работает — буду разбираться.

Заработало. `build.gradle.kts` случайно вложил под `src`.

File: java-tests/vanilla/src/test/kotlin/io/spine/validation/java/GoesConstraintTest.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    fun `if associated field is not set and the target field is set 'a violation is produced'`() {
+        val paper = Paper.newBuilder()
+            .setWhenArchived(Time.currentTime())
+            .buildPartial()
+
+        val error = paper.validate()
+        assertThat(error).isPresent()
+
+        val violations = error.get().getConstraintViolationList()
+        Truth.assertThat(violations)
```
Comment (dmdashenkov) [inline java-tests/vanilla/src/test/kotlin/io/spine/validation/java/GoesConstraintTest.kt:60]: Please add a static import.

**Review by dmdashenkov — APPROVED** [review]
LGTM with a couple of suggestions.
